features/machine/machine.feature

```diff
@@ -78,3 +78,111 @@ Feature: Machine features testing
     # node should be deleted
     Given I switch to cluster admin pseudo user
     Then I wait for the resource "node" named "<%= cb.new_node %>" to disappear within 600 seconds
+
+  # @author jhou@redhat.com
+  # @case_id OCP-25605
+  @admin
+  @destructive
+  Scenario: Machine Phase Lifecycle
+    Given I have an IPI deployment
+    And I store all machinesets to the :machinesets clipboard
+
+    Given evaluation of `machine_set.desired_replicas` is stored in the :replicas_original clipboard
+    And evaluation of `machine_set.desired_replicas.to_i + 1` is stored in the :replicas_expected clipboard
+
+    # scale up
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= machine_set.name %>     |
+      | replicas | <%= cb.replicas_expected %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+
+    # register clean-up just in case of failure
+    Given I register clean-up steps:
+    """
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= machine_set.name %>     |
+      | replicas | <%= cb.replicas_original %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+    """
+
+    Given I store the last provisioned machine in the :new_machine clipboard
+
+    # Phase "Provisioning"
+    When I run the :get admin command with:
+      | resource      | machine               |
+      | resource_name | <%= cb.new_machine %> |
+      | n             | openshift-machine-api |
+    Then the output should contain:
+      | Provisioning |
```

Comment:
@jhou1 I'm not sure how reliable the test can pass with this check...  From my experience that state comes and goes pretty quickly so it's hard to catch that state.  Perhaps skip this check?

Reply:
Provisioning might take some time, not sure about `Provisioned`. My suggestion would be to just wait for some time until machine becomes Running and when removing - wait some time to see machine number becomes what we want.

Testing all different statuses a machine would go through IMO falls outside blackbox testing.

Maybe we need steps like:
```
When machine set ... is upscaled with 1 more machine
And I register clean-up steps
"""
When machine set ... is downscaled with 1 machine
Then machine set ... has the expected number of machines and they become running
"""
Then machine set ... has the expected number of machines and they become running
```

This will simplify such scenarios greatly. If we don't need a lot of scaling maybe not worth it. You decide. But I still support Peter, thant checking transient statuses is potentially problematic as we saw with pods and builds in the past. Please avoid if this is not an important feature that we need to test for out customers.

Reply:
btw we can add a method `#ready?` to the MachineSet class and use this step to check machine is running:
```
/^(I|admin) waits? for the#{OPT_QUOTED} (\w+) to become ready(?: in the#{OPT_QUOTED} project)?(?: up to (\d+) seconds)?$/ 
```

Reply:
@akostadinov I can add that step(s) if @jhou1 is busy with other stuff

Reply:
@akostadinov @pruan-rht You are right, this test would probably be unreliable. The way how the machine phase is tested, can not ensure 100% pass ratio, especially when the cluster has pressure and api response is slow. I'll remove it, thanks